packages/common/src/utils/get-hub-request-options.ts

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2018 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+
+import { UserSession } from "@esri/arcgis-rest-auth";
+import { getSelf } from "@esri/arcgis-rest-portal";
+import { getHubApiUrlFromPortal } from "../urls/get-hub-api-url-from-portal";
+import { getPortalApiUrl } from "../urls/get-portal-api-url";
+import { IRequestOptions } from "@esri/arcgis-rest-request";
+import { IHubRequestOptionsPortalSelf } from "../types";
+
+export interface IGetHubRequestOptionsOptions extends IRequestOptions {
```

Comment:
Not a fan of the `...OptionsOptions` interface name, but felt `...OptionsParams` was equally ugly, LMK if you have any strong feelings about this interface name.